# PT001

`use @pytest.fixture over @pytest.fixture()`

## Configuration

* `pytest-fixture-no-parentheses`  
Boolean flag specifying whether `@pytest.fixture()` without parameters
should have parentheses.  
If the option is set to true (the default), `@pytest.fixture` is valid
and `@pytest.fixture()` is an error.  
If set to false, `@pytest.fixture()` is valid and `@pytest.fixture` is
an error.

## Examples

Bad code (assuming `pytest-fixture-no-parentheses` set to true):

```python
import pytest

@pytest.fixture()
def my_fixture():
    ...
```

Good code:

```python
import pytest

@pytest.fixture
def my_fixture():
    ...
```

## Rationale

* to enforce consistency between all fixtures in a codebase
