# PT029

`set the expected warning in pytest.warns()`

## Examples

Bad code:

```python
import pytest

def test_foo():
    with pytest.warns():
        do_something()
```

Good code:

```python
import pytest

def test_foo():
    with pytest.warns(SomeWarning):
        do_something()
```


## Rationale

* not passing the warning class will fail at runtime
