# PT023

`useless pytest.mark.usefixtures without parameters`

## Examples

Bad code:

```python
import pytest

@pytest.mark.usefixtures()
def test_something():
    ...

@pytest.mark.usefixtures
def test_something_else():
    ...
```

Good code:

```python
import pytest

@pytest.mark.usefixtures('a')
def test_something():
    ...
```

## Rationale

* such mark has no effect and is unnecessary
