# PT027

`use pytest.raises() instead of unittest-style '{assertion}'`

## Examples

Bad code:

```python
import unittest

class TestFoo(unittest.TestCase):
    def test_foo(self):
        with self.assertRaises(ValueError):
            raise ValueError('foo')
```

Good code:

```python
import pytest
import unittest

class TestFoo(unittest.TestCase):
    def test_foo(self):
        with pytest.raises(ValueError):
            raise ValueError('foo')
```

## Rationale

* to enforce the assertion style recommended by pytest
