# PT024

`pytest.mark.asyncio is unnecessary for fixtures`

## Examples

Bad code:

```python
import pytest

@pytest.mark.asyncio()
@pytest.fixture()
async def my_fixture():
    return 0
```

Good code:

```python
import pytest

@pytest.fixture()
async def my_fixture():
    return 0
```

## Rationale

* the mark is useless on fixtures and therefore unnecessary
