# PT028

`test function {name} has default value for argument {arg}, remove it`

## Examples

Bad code:

```python
def test_foo(bar=42):
    pass
```

Good code:

```python
def test_foo(bar):
    pass
```

## Rationale

* even if the corresponding fixture is defined, current behavior of pytest is 
   to use the default value instead of injecting the fixture
* see original pytest issue: https://github.com/pytest-dev/pytest#12693
