# PT032

`pytest.{name}() should be used in its context-manager form`

## Examples

Bad code:
```python
import pytest
import warnings

def my_raise_warning():
    warnings.warn(DeprecationWarning, "blah")

def test_foo():
    excinfo = pytest.raises(ValueError, int, "notanumber")
    excinfo.match("invalid literal")

def test_bar():
    warninfo = pytest.warns(DeprecationWarning, my_raise_warning)
    warninfo.match("blah")
    warninfo = pytest.deprecated_call(my_raise_warning)
    warninfo.match("blah")
```

Good code:

```python
import pytest

def test_foo():
    with pytest.raises(ValueError, match="invalid literal"):
        int("notanumber")

def test_bar()
    with pytest.warns(DeprecationWarning, match="blah"):
        warnings.warn(DeprecationWarning, "blah")
    with pytest.deprecated_call():
        warnings.warn(DeprecationWarning, "blah")
```

## Rationale
The context-manager form is easier to read and extend, allows for directly specifying `match`, and is much more widely used.
